lib/services/types.proto

```diff
@@ -766,13 +791,23 @@ message RoleOptions {
     // MaxSessions defines the maximum number of
     // concurrent sessions per connection.
     int64 MaxSessions = 10 [ (gogoproto.jsontag) = "max_sessions,omitempty" ];
+
+    // RequestAccess defines the access request stategy (optional|note|always)
+    // where optional is the default.
+    string RequestAccess = 11 [
+        (gogoproto.jsontag) = "request_access,omitempty",
+        (gogoproto.casttype) = "RequestStrategy"
+    ];
```

Comment:
Protobufs support a enum type.

https://github.com/gravitational/teleport/blob/master/lib/services/types.proto#L25-L32

Reply:
Those serialize to integers by default, and this field needs to be a string when in json or yaml format since users are expected to manually modify the field.  I could write a custom marshaler for it, but IMO the benefit is negligible.  Especially since go doesn't have enums, so we end up with a bunch of constants under the hood anyhow.

Reply:
Okay, makes sense.